### ISSUE TITLE
Installation error

### ISSUE THREAD

**Issue by yuval** [stanford-futuredata/ColBERT#183]
On Ubuntu. I ran:

```
$ conda env create -f conda_env.yml
$ conda activate colbert
```

It finished successfully it seems. But then I would get the following error:

```
$ python -c 'import torch'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/.../anaconda3/envs/colbert/lib/python3.8/site-packages/torch/__init__.py", line 229, in <module>
    from torch._C import *  # noqa: F403
ImportError: /.../anaconda3/envs/colbert/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so: undefined symbol: cudaGraphInstantiateWithFlags, version libcudart.so.11.0
```

I noticed conda installed the new pytorch 2 so thought that might be the problem (I'm using cuda 11.8 so it should be OK). I downgraded to 1.13.1:

```
$ conda install pytorch==1.13.1 torchvision==0.14.1 torchaudio==0.13.1 -c pytorch
```

After this it worked. However, it didn't recognize the GPU and indexing the data with CPU is extremely slow. 

I ran 'conda list' and found out I installed the pytorch cpu version. I then updated it to the gpu version and things seem to work fine now:

```
$ conda install -c conda-forge pytorch-gpu
```
(you can instead update the conda_env.yml file with the versions you want and create a new env or update an existing one).

Another issue is that the installation doesn't play well with Mac (M1/M2) cause of Cuda. It'd be nice to provide installation  that supports GPU on Mac.

**In case it helps**

Here's the cuda/torch output of 'conda list' after 'conda env create -f conda_env.yml' and before my other changes:

```
cuda-cudart               11.8.89                       0    nvidia
cuda-cupti                11.8.87                       0    nvidia
cuda-libraries            11.8.0                        0    nvidia
cuda-nvrtc                11.8.89                       0    nvidia
cuda-nvtx                 11.8.86                       0    nvidia
cuda-runtime              11.8.0                        0    nvidia
cudatoolkit               11.3.1               ha36c431_9    nvidia
faiss-gpu                 1.7.3           py3.8_h28a55e0_0_cuda11.3    pytorch
libfaiss                  1.7.3           hfc2d529_0_cuda11.3    pytorch
pytorch                   2.0.0           py3.8_cuda11.8_cudnn8.7.0_0    pytorch
pytorch-cuda              11.8                 h7e8668a_3    pytorch
pytorch-mutex             1.0                        cuda    pytorch

faiss-gpu                 1.7.3           py3.8_h28a55e0_0_cuda11.3    pytorch
ffmpeg                    4.3                  hf484d3e_0    pytorch
libfaiss                  1.7.3           hfc2d529_0_cuda11.3    pytorch
pytorch                   2.0.0           py3.8_cuda11.8_cudnn8.7.0_0    pytorch
pytorch-cuda              11.8                 h7e8668a_3    pytorch
pytorch-mutex             1.0                        cuda    pytorch
torchaudio                2.0.0                py38_cu118    pytorch
torchtriton               2.0.0                      py38    pytorch
torchvision               0.15.0               py38_cu118    pytorch
```

**Comment by yuval** [comment]
I spoke too soon. The indexing crashed after a while. Something about RuntimeError: Error building extension 'decompress_residuals_cpp'. I'm looking into it and might reach out for help soon.

**Comment by yuval** [comment]
TLDR the default installation failed because I think the current setup doesn't play well with Pytorch 2.0. I fixed it by specifying an older version in the conda env yml file:

```
  - pytorch=1.13.1
  - torchaudio=0.13.1
  - torchvision=0.14.1
```

**Comment by okhat** [comment]
Hmm, thank you @yuval ! I will edit the conda file. Do you think the issue is fundamental or just happens in your particular setup?

**Comment by yuval** [comment]
I believe it's not my setup. I got the same error on two different machines, both were created right before I tried this.
